skyportal/handlers/api/reminder.py

```diff
@@ -0,0 +1,900 @@
+        multiple:
+          description: Retrieve all reminders associated with specified resource
+          tags:
+            - reminders
+            - spectra
+            - sources
+            - gcn_event
+          parameters:
+            - in: path
+              name: associated_resource_type
+              required: true
+              schema:
+                type: string
+                enum: [sources]
+              description: |
+                 What underlying data the reminder is on, e.g., "sources"
+                 or "spectra" or "gcn_event".
+            - in: path
+              name: resource_id
+              required: true
+              schema:
+                type: string
+              description: |
+                 The ID of the underlying data.
+                 This would be a string for a source ID
+                 or an integer for other data types like spectrum or gcn_event.
+          responses:
+            200:
+              content:
+                application/json:
+                  schema: ArrayOfReminders
+            400:
+              content:
+                application/json:
+                  schema: Error
+        """
+        if reminder_id is None:
+            if associated_resource_type.lower() == "sources":
+                reminders = (
+                    Reminder.query_records_accessible_by(self.current_user)
+                    .filter(Reminder.obj_id == resource_id)
+                    .all()
+                )
+            elif associated_resource_type.lower() == "spectra":
+                reminders = (
+                    ReminderOnSpectrum.query_records_accessible_by(self.current_user)
+                    .filter(ReminderOnSpectrum.spectrum_id == resource_id)
+                    .all()
+                )
+            elif associated_resource_type.lower() == "gcn_event":
+                reminders = (
+                    ReminderOnGCN.query_records_accessible_by(self.current_user)
+                    .filter(ReminderOnGCN.gcn_id == resource_id)
+                    .all()
+                )
+            elif associated_resource_type.lower() == "shift":
+                reminders = (
+                    ReminderOnShift.query_records_accessible_by(self.current_user)
+                    .filter(ReminderOnShift.shift_id == resource_id)
+                    .all()
+                )
+            else:
+                return self.error(
+                    f'Unsupported associated resource type "{associated_resource_type}".'
+                )
+            self.verify_and_commit()
+            return self.success(data=reminders)
+
+        try:
+            reminder_id = int(reminder_id)
+        except (TypeError, ValueError):
+            return self.error("Must provide a valid (scalar integer) reminder ID. ")
+
+        # the default is to reminder on an object
+        if associated_resource_type.lower() == "sources":
+            try:
```

Comment:
You have a try expect for each of the if statement, but can't it just be replaced by one try except around all the if statements ?